src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -0,0 +1,119 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRIORITY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_RATING;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STATUS;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.ActiveListType;
+import seedu.address.model.book.Book;
+import seedu.address.model.book.Priority;
+import seedu.address.model.book.Rating;
+import seedu.address.model.book.Status;
+import seedu.address.model.book.exceptions.BookNotFoundException;
+import seedu.address.model.book.exceptions.DuplicateBookException;
+
+/**
+ * Changes the rating of an existing Book in the address book.
+ */
+public class EditCommand extends UndoableCommand {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the rating of the book identified "
+            + "by the index number used in the last book listing. "
+            + "Existing rating will be overwritten by the input.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_RATING + "-1" + PREFIX_PRIORITY + "low" + PREFIX_STATUS
+            + "unread";
+
+    public static final String MESSAGE_ADD_EDITING_SUCCESS = "Editing to Book: %1$s";
+    public static final String MESSAGE_DELETE_EDITING_SUCCESS = "Removed editing from Book: %1$s";
+    public static final String MESSAGE_WRONG_ACTIVE_LIST = "Cannot be edited.";
+
+    private final Index index;
+    private final Rating rating;
+    private final Priority priority;
+    private final Status status;
+    private Book bookToEdit;
+    private Book editedBook;
+
+    /**
+     * @param index of the book in the filtered book list to edit the rating
```

Comment:
Update this line. Add `@param` description for `priority` and `status`. Make sure they all end with a full-stop.